### PR TITLE
bpo-37021: Port _randommodule to the argument clinic

### PR DESCRIPTION
**Pull request by pablogsal**
https://bugs.python.org/issue37021

### CODE REVIEW

**Comment by pablogsal** [comment]
I have preserved the docstrings, please advise if you consider that you wish them changed to something else.

File: Modules/_randommodule.c [hunk 1]
```diff
@@ -317,8 +330,34 @@ random_seed(RandomObject *self, PyObject *args)
...
 }
 
+/*[clinic input]
+_random.Random.seed
+
+  self: self(type="RandomObject *")
+  n: object = None
+  /
+
+seed([n]) -> None.  Defaults to current time.
```
Comment (vstinner) [inline Modules/_randommodule.c]: Hum, this doc sounds outdated, no?

File: Modules/_randommodule.c [hunk 2]
```diff
@@ -231,21 +248,17 @@ random_seed_time_pid(RandomObject *self)
...
     PyObject *n = NULL;
     uint32_t *key = NULL;
     size_t bits, keyused;
     int res;
-    PyObject *arg = NULL;
-
-    if (!PyArg_UnpackTuple(args, "seed", 0, 1, &arg))
-        return NULL;
 
-     if (arg == NULL || arg == Py_None) {
```
Comment (ZackerySpytz) [inline Modules/_randommodule.c:247]: Why was this change made?

File: Modules/_randommodule.c [hunk 3]
```diff
@@ -231,21 +248,17 @@ random_seed_time_pid(RandomObject *self)
     init_by_array(self, key, Py_ARRAY_LENGTH(key));
 }
 
-static PyObject *
```
Comment (ZackerySpytz) [inline Modules/_randommodule.c]: Why was this change made?

**Review by vstinner — APPROVED** [review]
LGTM. Just a minor comment on an outdated doc.

File: Modules/_randommodule.c (hunk 2 above)
Comment (pablogsal) [reply Modules/_randommodule.c:247]: Because the indentation was wrong

**Review by vstinner — APPROVED** [review]
LGTM.

**Pull request merged**